src/cdk/table/table.ts

```diff
@@ -999,9 +1034,11 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
       throw getTableUnknownDataSourceError();
     }
 
-    this._renderChangeSubscription = dataStream!.pipe(takeUntil(this._onDestroy))
+    this._renderChangeSubscription = combineLatest([dataStream!, this.viewChange]).pipe(
+        map(([data, range]) => range ? (data || []).slice(range.start, range.end) : []),
```

Comment:
@jelbourn @andrewseguin This is a potentially breaking change. A couple unit tests assert that array data sources can be updated by reference and then call `renderRows()` to display the changes. Calling `slice()` on the data set changes the reference and breaks the tests, because the internal list of rows is no longer the same reference as the original data source.

Recommendations?

Reply:
Should we just add the slicing to `renderRows`? The range would need to either be passed in as an optional argument or stored on the table like how we store `this._data`

Reply:
Done